### DISCUSSION TITLE
The accuracy of the forecast vs using all or part of the historical data

### DISCUSSION THREAD

**Discussion by Msaleh87** [ourownstory/neural_prophet#1095] [Q&A - get help using NeuralProphet] [answered]
Hi,

This is my code:

```
m = NeuralProphet(n_forecasts = prediction_horizon,
    n_lags=int(16*24* timestep),
    yearly_seasonality="auto",
    weekly_seasonality="auto",
    daily_seasonality="auto", 
    growth="off", 
    seasonality_reg= 0.5, 
    optimizer = 'AdamW',
    learning_rate = 0.1, 
    n_changepoints="auto", 
    changepoints_range = "auto")

m = m.add_country_holidays("US", mode="additive") # I added an equal sign, check how would this impact the original code
m = m.add_future_regressor(name='A')
m = m.add_future_regressor(name='B')
m = m.add_future_regressor(name='C')
metrics = m.fit(N_Ph_Train, freq = freq)
m.highlight_nth_step_ahead_of_each_forecast(step_number=m.n_forecasts)

df_prdeict = N_Ph_data_train_resample_5min[: Threshold].copy()
future = m.make_future_dataframe(df = df_prdeict[-int(prediction_horizon+m.n_lags):],
                                 regressors_df=future_regressors_df , 
                                 periods = prediction_horizon, 
                                 n_historic_predictions = prediction_horizon)
forecast1 = m.predict(df=future)
```
Now, if I use `df = df_prdeict[-int(prediction_horizon+m.n_lags):]` or `df = df_prdeict` inside the` m.make_future_dataframe` The accuracy is exactly the same. Does that mean that the model uses only (prediction_horizon+m.n_lags) data points from the historical data to make the prediction?

Also, if I want to make the model adaptive to new acquired data points, I think I need to retrain the model from scratch every 4-10 hours (assuming that every data point is acquired every 5 mins) to have a reasonable amount of new data compared to the (prediction_horizon+m.n_lags) data points, correct? I did this image below, and this is my assumption regarding adaptability. Is it correct? My application takes a data point every 5mins and predicts the next 48 hours' horizon with 5mins resolution, it is more of an online application.

![NPH1](https://user-images.githubusercontent.com/42920113/210657170-88681144-41ab-452c-9aef-6253e9032c51.png)

**Comment by Msaleh87** [comment]
Hi @ourownstory,
What do you think?

**Reply by judussoari** [reply, marked as answer]
Hey @Msaleh87 you are correct, with autoregression enabled, we take n_lags datapoints as an input, to predict the next n_forecasts. 

Re 2nd question: I think both approaches are valid. If the data adaptively shows similar behavior, then the pre-trained model (on the non-updated data) is capable of doing forecasts. An adaptive approach would give new weights, and therefore potentially shows a more informed forecast on the new data.
